pkg/crc/machine/cert-renewal.go

```diff
@@ -140,105 +41,32 @@ func RegenerateCertificates(sshRunner *ssh.SSHRunner, machineName string) error
-		_, _, err := oc.RunOcCommand(
-			"patch", resourceType, "cluster",
-			fmt.Sprintf("--patch=%s", forceRedeploymentPatch),
-			"--type=merge")
-		if err != nil {
-			logging.Warnf("Error forcing redeployment of %s: %v", resourceType, err)
-			return err
-		}
-	}
-
-	_, err = sshRunner.Run(fmt.Sprintf("sudo KUBECONFIG=%s /usr/local/bin/recover-kubeconfig.sh >kubeconfig && sudo mv kubeconfig /etc/kubernetes/kubeconfig", recoveryPod.GetKubeconfigPath()))
-	if err != nil {
-		logging.Debugf("Error invoking recover-kubeconfig.sh: %v", err)
-		return err
-	}
-
-	output, _, err := oc.RunOcCommand("get", "configmap", "kube-apiserver-to-kubelet-client-ca",
-		"-n openshift-kube-apiserver-operator",
-		"--template='{{ index .data \"ca-bundle.crt\" }}'")
-	if err != nil {
-		logging.Debugf("Error reading kubelet client CA: %v", err)
-		return err
-	}
-	/* Copy the ca-bundle data we just read to etc/kubernetes/kubelet-ca.crt */
-	err = sshRunner.SetTextContentAsRoot("/etc/kubernetes/kubelet-ca.crt", output, 0644)
-	if err != nil {
-		logging.Debugf("Error writing kubelet client CA to /etc/kubernetes/kubelet-ca.crt: %v", err)
-		return err
-	}
-
-	_, _ = sshRunner.Run("sudo touch /run/machine-config-daemon-force")
-
-	_, err = sd.Stop("kubelet")
-	if err != nil {
-		return err
-	}
-	_, _ = sshRunner.Run("sudo rm -rf /var/lib/kubelet/pki /var/lib/kubelet/kubeconfig")
-	_, _ = sshRunner.Run("sudo crictl stopp $(sudo crictl pods -q)")
-	_, _ = sshRunner.Run("sudo crictl rmp $(sudo crictl pods -q)")
-
-	_, err = sd.Start("kubelet")
-	if err != nil {
-		logging.Debugf("Error restarting kubelet service: %v", err)
-		return err
-	}
-
+	oc := oc.UseOCWithConfig(machineName)
 	/* 2 CSRs to approve, one right after kubelet restart, the other one a few dozen seconds after
-	approving the first one */
+	approving the first one
+	- First one is requested by system:serviceaccount and address to node-bootstrapper
+	- Second one is requested by system:node and address to node-name*/
 	err = waitForPendingCsrs(oc)
 	if err != nil {
-		logging.Debugf("Error waiting for first pending CSR: %v", err)
+		logging.Debugf("Error waiting for first pending (node-bootstrapper) CSR: %v", err)
 		return err
 	}
 	err = oc.ApproveNodeCSR()
 	if err != nil {
-		logging.Debugf("Error approving first pending CSR: %v", err)
+		logging.Debugf("Error approving first pending (node-bootstrapper) CSR: %v", err)
 		return err
 	}
 
 	err = waitForPendingCsrs(oc)
 	if err != nil {
-		logging.Debugf("Error waiting for second pending CSR: %v", err)
+		logging.Debugf("Error waiting for second pending (system:node) CSR: %v", err)
 		return err
 	}
 	err = oc.ApproveNodeCSR()
 	if err != nil {
-		logging.Debugf("Error approving second pending CSR: %v", err)
+		logging.Debugf("Error approving second pending (system:node) CSR: %v", err)
 		return err
 	}
 
-	/* The cluster needs to settle for a few minutes before connections to
-	 * api.crc.testing:6443 gets a valid TLS certificate
-	 */
```

Comment:
With the automated cert recovery, the apiserver is immediatly available?

Reply:
@cfergeau Here we don't create the recovery apiserver which need to be deleted so our current apiserver handle it and we don't need to now check for default  api connections.

Reply:
This was not only checking if we can reach the api server, this was also checking if the certificates of the apiserver are valid. Actually I'm not sure how this all works, the certificate for the api.crc.testing:6443 connection are renewed automatically, so that we can approve the CSR for a different set of certificates?

Reply:
@cfergeau before the recovery cert server used to run on `7443` port so we need to delete that recovery pod after certs are created and csr are approved and then we used this to make sure our default apiserver which runs on `6443` port able to comeup, now we don't need that different recovery pod so this step also not needed.